### PR TITLE
Many document fixes

### PR DESCRIPTION
**Pull request by yuyichao**
- `TCP*` rename
- A few missing quote in the document
- Split the doc for `unique` for different signatures and update their signatures in the doc. (Also move the doc strings inline...) (Ref https://github.com/JuliaLang/julia/pull/13622/)
- Do not print `Unexported method missing doc` in `genstdlib.jl` since we don't really look at it anyway (and they hide more important warnings...)....
- Remove document of the `Pkg.*` functions that are now in `PkgDev`.
- Try to update the package development manual for `PkgDev`.

c.c. @tkelman 

Close https://github.com/JuliaLang/julia/issues/13697 ?

### CODE REVIEW

**Comment by tkelman** [comment]
backport for the parts of the first commit that apply to release-0.4 (not the new `unique` method but most of the rest)

File: base/multidimensional.jl [hunk 1]
```diff
@@ -784,6 +784,11 @@ immutable Prehashed
 end
 hash(x::Prehashed) = x.hash
 
+doc"""
+    unique(itr[, dim])
+
+Returns an array containing only the unique elements of the iterable `itr`, in the order that the first of each set of equivalent elements originally appears. If `dim` is specified, returns unique regions of the array `itr` along `dim`.
```
Comment (tkelman) [inline base/multidimensional.jl]: line wrap?

File: doc/manual/packages.rst [hunk 2]
```diff
@@ -351,6 +351,7 @@ where ``USERNAME`` is your actual GitHub user name.
 Once you do this, the package manager knows your GitHub user name and can configure things accordingly.
 You should also `upload <https://github.com/settings/ssh>`_ your public SSH key to GitHub and set up an `SSH agent <http://linux.die.net/man/1/ssh-agent>`_ on your development machine so that you can push changes with minimal hassle.
 In the future, we will make this system extensible and support other common git hosting options like `BitBucket <https://bitbucket.org>`_ and allow developers to choose their favorite.
+Since the package development functions has been moved to the `PkgDev <https://github.com/JuliaLang/PkgDev.jl>`_ package, you need to run ``Pkg.add("PkgDev"); import PkgDev`` to access the functions starts with ``PkgDev.`` in the document below.
```
Comment (tkelman) [inline doc/manual/packages.rst]: "to access the functions starting with"

File: doc/stdlib/pkg.rst [hunk 3]
```diff
@@ -7,6 +7,8 @@
 All package manager functions are defined in the ``Pkg`` module. None of the ``Pkg`` module's functions are exported;
 to use them, you'll need to prefix each function call with an explicit ``Pkg.``, e.g. ``Pkg.status()`` or ``Pkg.dir()``.
 
+Functions for package development (e.g. ``tag``, ``publish``, etc.) have been moved to the `PkgDev <https://github.com/JuliaLang/PkgDev.jl>`_ package. See `PkgDev README <https://github.com/JuliaLang/PkgDev.jl/blob/master/README.md#usage>`_ for the document of those functions.
```
Comment (tkelman) [inline doc/stdlib/pkg.rst]: "for the documentation of those functions"

**Comment by yuyichao** [comment]
Done. (rerun `genstdlib` to make sure nothing else changes...)

**Comment by jakebolewski** [comment]
lgtm

**Pull request merged**